### PR TITLE
[sdn_tests]: Adding Build support for pins_ondatra.

### PR DESCRIPTION
**Pull request by VSuryaprasad-HCL**
### Description of PR
[sdn_tests]: Adding Build support for pins_ondatra.

Summary:
- Adding Build support for pins_ondatra.

Build Result:

       INFO: Analyzed 0 targets (1 packages loaded, 0 targets configured).
       INFO: Found 0 targets...
       INFO: Elapsed time: 0.746s, Critical Path: 0.02s
       INFO: 1 process: 1 internal.
       INFO: Build completed successfully, 1 total action

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [-] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311

### CODE REVIEW

File: sdn_tests/pins_ondatra/pins_deps.bzl [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+            strip_prefix = "rules_proto-3f1ab99b718e3e7dd86ebdc49c580aa6a126b1cd",
+            sha256 = "c9cc7f7be05e50ecd64f2b0dc2b9fd6eeb182c9cc55daf87014d605c31548818",
+        )
+    if not native.existing_rule("sonic_swss_common"):
+        # We use `git_repository` over `http_archive` only because this is a private repo
+        # requiring SSH authentication.
+        git_repository(
+            name = "sonic_swss_common",
+            commit = "672b1cfe1914489b79a91fff8a539b6c195f8959",
+            remote = "git@github.com:pins/sonic-swss-common.git",
```
Comment (kishanps) [inline sdn_tests/pins_ondatra/pins_deps.bzl:201]: @parth-opensrc What would be the sonic-net equivalent commit id ? We can't use PINS refpt here.

Comment (parth-opensrc) [reply sdn_tests/pins_ondatra/pins_deps.bzl:201]: I think this repo may not be needed for ondatra, so can be removed for now.
I will clean up this file in future.

Comment (kishanps) [reply sdn_tests/pins_ondatra/pins_deps.bzl:201]: @VSuryaprasad-HCL Can you remove this and re-attach the pass build logs.

Comment (VSuryaprasad-HCL) [reply sdn_tests/pins_ondatra/pins_deps.bzl]: I have addressed the changes and reattached the build log.

File: sdn_tests/pins_ondatra/infra_deps.bzl [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+    git_repository(
+        name = "com_google_googleapis",
+        remote = "https://github.com/googleapis/googleapis",
+        commit = "c4915db59896a1da45b55507ece2ebc1d53ef6f5",
+        shallow_since = "1642638275 -0800",
+    )
+
+    go_repository(
+        name = "com_github_jstemmer_go_junit_report_v2",
+        importpath = "github.com/jstemmer/go-junit-report/v2",
```
Comment (kishanps) [inline sdn_tests/pins_ondatra/infra_deps.bzl:313]: @VSuryaprasad-HCL Getting a 404 error on this url ? Maybe we haven't added the build targets that exercise this to cause a failure ?
cc: @parth-opensrc

File: sdn_tests/pins_ondatra/infra_deps.bzl [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+        sum = "h1:BVBb1o0TfOuRCMykVAYJ1r2yoZ+ByE0f19QNF4ngQ0M=",
+        version = "v2.0.1-0.20220823220451-7b10b4285462",
+    )
+
+    go_repository(
+        name = "com_github_patrickmn_go_cache",
+        importpath = "github.com/patrickmn/go-cache",
```
Comment (kishanps) [inline sdn_tests/pins_ondatra/infra_deps.bzl:320]: @vamsipunati, @parth-opensrc 
This and the previous one, are we ok to include these as dependencies in sonic-net ?

Comment (parth-opensrc) [reply sdn_tests/pins_ondatra/infra_deps.bzl:320]: Ondatra needs this to be built.
https://github.com/openconfig/ondatra/blob/main/internal/ixgnmi/ixgnmi.go#L39, and 
https://github.com/openconfig/ondatra/blob/main/report/report_test.go#L23

We are using checksum, so it should be fine, I guess.

File: sdn_tests/pins_ondatra/infra_deps.bzl (hunk 2 above)
Comment (parth-opensrc) [reply sdn_tests/pins_ondatra/infra_deps.bzl:313]: Try removing it, if the build works, it should be fine to remove the dep.
If it doesn't, you can try changing the rule:
```
go_repository(
    name = "com_github_jstemmer_go_junit_report_v2",
    commit = "add_commit_hash", 
    importpath = "github.com/jstemmer/go-junit-report/v2",
    remote = "https://github.com/jstemmer/go-junit-report",
    vcs = "git"
)
```

Comment (VSuryaprasad-HCL) [reply sdn_tests/pins_ondatra/infra_deps.bzl:313]: Tried removing the dep, The build was successful. Attaching the build info in the chat box.

         INFO: Analyzed 0 targets (1 packages loaded, 0 targets configured).
         INFO: Found 0 targets...
         INFO: Elapsed time: 2.596s, Critical Path: 0.02s
         INFO: 1 process: 1 internal.
         INFO: Build completed successfully, 1 total action

File: sdn_tests/pins_ondatra/infra_deps.bzl (hunk 3 above)
Comment (VSuryaprasad-HCL) [reply sdn_tests/pins_ondatra/infra_deps.bzl:320]: @parth-opensrc Tried removing the rules, The build was successful. Attaching the build info in the chat box 

       INFO: Analyzed 0 targets (1 packages loaded, 0 targets configured).
       INFO: Found 0 targets...
       INFO: Elapsed time: 0.593s, Critical Path: 0.00s
       INFO: 1 process: 1 internal.
       INFO: Build completed successfully, 1 total action

**Review by rkavitha-hcl — APPROVED** [review]

**Review by vamsipunati — APPROVED** [review]

**Comment by VSuryaprasad-HCL** [comment]
https://github.com/sonic-net/sonic-mgmt/pull/11771

**Pull request merged**